app/controllers/import/configurations_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Import::ConfigurationsController < ApplicationController
+  layout "imports"
+
+  before_action :set_import
+
+  def show
+  end
+
+  def update
+    @import.update!(import_params)
+    @import.generate_rows_from_csv
+    @import.reload.sync_mappings
```

Comment:
I'll cover the "service object" for all of the comments here.  In general, I agree that this could be cleaned up a bit, but rather than a service object, I'd prefer to see it directly in the model as something like `@import.update_and_load!` to fit the domain a bit better.

Overall, I think this feature may be a bit at odds with introducing a DI service object mainly because of the inheritance structure:

```rb
class Import < ApplicationRecord
class TransactionImport < Import
class TradeImport < Import
class AccountImport < Import
class MintImport < Import
```

Each of these subclasses have a specific `import!` implementation and type-specific steps / columns.  If we introduced a service object, we'd still have to handle this inheritance; it would just be through something like `TransactionImporter`, `TradeImporter`, `AccountImporter`, and `MintImporter` and then we'd be removing these subclasses entirely (as they'd be pretty anemic at that point).

If the import logic gets a lot more complex in the future, I think there could be an argument for adding an additional layer of abstraction specifically for the `import!` method, but for now I can't see a strong argument for that given how thin each of these subclasses are.